segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/SegmentMetadataUpdateTransaction.java

```diff
@@ -474,15 +502,18 @@ private void preProcessAttributes(AttributeUpdateCollection attributeUpdates)
             return;
         }
 
+        // Make sure that the number of existing ExtendedAttributes + incoming ExtendedAttributes does not exceed the limit.
+        if (NameUtils.isTransientSegment(this.name) && getUniqueExtendedAttributesCount(attributeUpdates) > TRANSIENT_ATTRIBUTE_LIMIT) {
```

Comment:
`NameUtils.isTransientSegment` is  very expensive method to run here. Use `metadata.getType().isTransientSegment`